tests/text_preparation/test_text_preparation_pipeline.py

```diff
@@ -137,3 +137,38 @@ def test_construct_pipeline_components():
         assert isinstance(normalizer, Normalizer)
     assert isinstance(text_preparation_pipeline.tokenizer, SpacyTokenizer)
     assert isinstance(text_preparation_pipeline.stemmer, EnglishNLTKStemmer)
+
+
+def test_construct_pipeline_components_invalid_input():
+
+    with pytest.raises(TypeError):
+        TextPreparationPipelineFactory.create_text_preparation_pipeline(
+            preprocessors = None,
+            normalizers = ("SpacyTokenizer"),
+            tokenizer = None,
+            stemmer = None
+        )
+
+    with pytest.raises(TypeError):
+        TextPreparationPipelineFactory.create_text_preparation_pipeline(
+            preprocessors = None,
+            normalizers = None,
+            tokenizer = "NoOpNormalizer",
+            stemmer = None
+        )
+    
+    with pytest.raises(TypeError):
+        TextPreparationPipelineFactory.create_text_preparation_pipeline(
+            preprocessors = None,
+            normalizers = None,
+            tokenizer = None,
+            stemmer = "NoOpPreprocessor"
+        )
+
+    with pytest.raises(TypeError):
+        TextPreparationPipelineFactory.create_text_preparation_pipeline(
+            preprocessors = ("NoOpNormalizer"),
```

Comment:
Could you also test passing in a mix, ie string and component type?

Reply:
Sure, so as in the example in test_construct_pipeline_components_valid_input() but with string/object components that are of an invalid type?

Reply:
Added